Composer/packages/client/src/pages/home/CardWidget.tsx

```diff
@@ -74,10 +76,29 @@ export const CardWidget: React.FC<CardWidgetProps> = ({
     const rect = e.currentTarget.getBoundingClientRect();
     const aspectRatio = rect.width / rect.height;
     if (aspectRatio < 1.5) {
+      setOddImageSize(true);
+    }
+    caclImageContainerRect();
+  };
+
+  // detect image container dimention, if it's too wide, apply background.
+  // Why not always put a background behind image, because when image size is nearly to container size, we don't want see 1px or 5px black line.
+  const caclImageContainerRect = () => {
+    const containerRect = imageContainerEl.current?.getBoundingClientRect();
```

Comment:
calling this function will make browser do a reflow, now imagine calling it on every pixel resize change. It's expensive. Is there a better way for doing this?

Reply:
@zhixzhan This should not a fix in Javascript. Inside a media query you need to do a display none or block on the image background.